### DISCUSSION TITLE
The difference of mass scaling between Abaqus and Openradioss

### DISCUSSION THREAD

**Discussion by QuatorzeM** [OpenRadioss/OpenRadioss#1775] [Q&A] [answered]
Dear community,

Recently, I have converted some .inp files of Abaqus to .rad files. For the mass scaling part, I found it's not so direct and some difference that may interest you. 

Here is the details:
1. for Abaqus, the process of mass scaling seems to be mainly conducted in elements. The characteristic time increment dt of every element is calculated then if the dt is smaller than the criterion, mass will be added firtly at the element and then distribute to the nodes. 
2. for Openradioss, following the logic of Abaqus, I firstly check the key word **/DT/Eltyp/Keywords/Iflag** , but I found it is used to change to small strain formulation to maintain the constant time step not the same idea as in Abaqus. Secondly, I checked **/DT/NODA/Kwyword3/Iflag** it's used for mass scaling. But the characteristic time increment dt  is calculated at each node, which confused me a little (in explicit dynamics, for element, we have dt = characteristic length / wave speed). Then I checked code, I found in code: DT(I) = MS(I) / STIFN(I) where MS sholuld be the nodal mass, STIFN is the nodal stiffness. For starter part, MS and STIFN should be the estimated value, in engine it should be the exact value.

I'd like to know, is my understanding right? and how the STIFN should be calculated?
Another important question is, if the method of mass scaling calculation is different,  how we should convert the mass scaling from Abaqus to Openradioss ?

Best regards,
Fabien

**Comment by qiangRadioss** [comment, marked as answer]
Hi Fabien,

yes, /DT/NODA/CST/Iflag should be used for mass scaling.
Besides the elementary time step, OpenRadioss has also another (optional)one: nodal time step which Abaqus and Ls-dyna have not.
there is 2 main advantages of this nodal time step :
1) generally nodal time step >= elementary time due to the fact that nodal one uses global assembled (at node) terms instead of local elementary terms, and nodal time step (almost all elements in Radioss) is computed  based on the elementary one (stiffness&mass are then distributed to nodes); you can find the same time step value when the model contains only one element.
2)time step compute involving contact, rigid body... can be easily estimated by nodal time step, that's why in OpenRadioss, when penalty contact happened during the run, elementary time step (if initially) will passe to nodal time step automatically.
For your question how to convert Abaqus mass scaling to OpenRadioss, i think converting the same target time step value to OpenRadioss could be already fine, then OpenRadioss will add less masses if they have the same elementary time step. (if OpenRadioss added more masses, there is a big chance that elementary time step is higher than Radioss' one)

hope this helps.

Qiang.

**Reply by QuatorzeM** [reply]
Hi Qiang, 

Thanks a lot. That's very helpful, which confirms my understanding.

I tried to use the same value as abaqus, but there is an energy error. I think maybe there are something wrong with the material part (it's a hyperelastic one) or the unit. I will check it.

Best regards,
Fabien

**Reply by MarianBulla** [reply]
Hello Fabien,
thank you very much for your interest in this topic.

Can you contact me at [communitymanager@openradioss.org](mailto:communitymanager@openradioss.org), please ?

Thanks and best regards,
Marian

**Reply by QuatorzeM** [reply]
Hello Marian,

It's my pleasure to receive your invitation :)

I am sorry that there days, I was busy moving to a new apartment. 

I will send you the mail tonight.

Have a nice day ！

Fabien
